ELEC 390/app/src/main/java/com/example/mitchelllichocki/elec390project/MapDisplayActivity.java

```diff
@@ -97,14 +108,30 @@ public void run(){
         setBeacon.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                if(beaconMarker != null){
-                    beaconMarker.remove();
-                }
-                if(!((lat > 90) || (lat < 0) || (Math.abs(lon) > 180))){
-                    beaconMarker = map.addMarker(new MarkerOptions().position(new LatLng(lat, lon)).title("Beacon").snippet("Lat: " + lat + "\n" + "Long: " + lon));
-                    backgroundWorker.setBeacon(map, username, childSelected, lat, lon, radius);
-                    lat = 1000;
-                    lon = 1000;
+                //savePosition();
+                if(savedPosition != null){
+                    //save in the database instead
+                    //BackgroundWorker backgroundWorker = new BackgroundWorker();
+                    //backgroundWorker.seBeacon(lat, lon);
+
+                    // this code only allows one saved region at a time
+                    if (mCircles != null){mCircles.clear();}
+                    map.clear();
+                    savedRegion = new DraggableCircle(savedPosition, savedRadius); // default radius is 50 meters
+                    savedRegion.hideMarkers();
+                    mCircles.add(savedRegion);
+                    //making sure we only save the same reagion once
+                    savedPosition = null;
+                    savedRadius = 50.0;
```

Comment:
Does this mean your Radius will always be 50 and you don't have a beacon slider or size modifier?